cpp/search/search.cpp

```diff
@@ -2003,7 +2003,10 @@ void Search::initNodeNNOutput(
   }
   if(isRoot && searchParams.rootNumSymmetriesToSample > 1) {
     vector<shared_ptr<NNOutput>> ptrs;
+    std::array<int, NNInputs::NUM_SYMMETRY_COMBINATIONS> symmetryIndexes {0,1,2,3,4,5,6,7};
```

Comment:
this feels strange to me, with NUM_SYMMETRY_COMBINATIONS being a variable, but the elements being hardcoded. std::iota ?

Reply:
Yeah I can change this to populate it with a loop if you prefer.

Reply:
I think it's fine, as long as it has the property of failing to compile if NNInputs::NUM_SYMMETRY_COMBINATIONS were to ever somehow change.

Reply:
(i.e. the critical thing is that we are never accidentally in a state where the two become inconsistent and compile without knowing it - either they should automatically be consistent, or else it should be the case that they aren't, but the inconsistent state fails to compile so that it cannot be accidentally missed).

Reply:
I like Sander's idea to populate it with std::iota. Is it just this?

std::iota(symmetryIndexes.begin(), symmetryIndexes.end(), 0);

Reply:
Change made.